### ISSUE TITLE
Sync Tags from Firefox Bookmarks 

### ISSUE THREAD

**Issue by arvidortwig** [marcelklehr/floccus#615]
**Is your feature request related to a problem? Please describe.**
When i upload the bookmarks to nextcloud, the folders get synced but not the tags. Using the Nextcloud Bookmarks App (via F-Droid) it shows only the tags not the folders. In Nextcloud also not tags are shown. 

**Describe the solution you'd like**
It would be great if the tags get synced to. 

**Describe alternatives you've considered**
Alternatively the Nextcloud Bookmark app should show the folders as well.

**Additional context**

**Comment by marcelklehr** [comment]
Hey,

thank you for your feedback! The situation here is indeed suboptimal, in flux and possibly a little confusing. There's multiple aspects to this:

- The bookmarks android app has [an open feature request for folders](https://gitlab.com/bisada/OCBookmarks/-/issues/25)
- Floccus [cannot access](https://github.com/marcelklehr/floccus/issues/28) Firefox's tags, as of yet.
- The server-side bookmarks app is installable on mobile as a progressive web app and thus gives you both tags and folders on mobile
- Firefox will probably soon [support floccus](https://github.com/marcelklehr/floccus/issues/3), obviating the need for an android app

Does this help you?

**Comment by arvidortwig** [comment]
Hey Marcel,

thanks for your reply. I left a comment with the owner of the Bookmarks App, asking for folder support. We will see if he responds. Would be a killer feature, as you already said ;) Regarding the progressive web app: Until now I only was able to "install" a progressive webapp (pwa) through chromium on my Desktop/Gnome. Never did this on a mobile. Was looking for it on android with fennec and chromium (LineageOS Standard Browser) but found no "button". Is the pwa installable through any browser ?
Thanks,
Arvid

**Comment by marcelklehr** [comment]
> Is the pwa installable through any browser ?

it should be: https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Add_to_home_screen

**Comment by github-account1111** [comment]
>Firefox will probably soon https://github.com/marcelklehr/floccus/issues/3, obviating the need for an android app

Are you saying the app will be deprecated in lieu of Firefox supporting floccus?
I think you may have to consider that

- on Android, most people use Chrome
- installing a second browser engine increases attack surface which the more security conscious users try to avoid
- Firefox is [less secure than Chromium](https://madaidans-insecurities.github.io/firefox-chromium.html) in general which ties back into the previous point

**Comment by marcelklehr** [comment]
> Are you saying the app will be deprecated in lieu of Firefox supporting floccus?

The post you are referring to is two years old, the floccus android app is less than a year old, after I realized that floccus support for Firefox Android isn't going to happen any time soon. I will not deprecate the floccus Android app.

**Comment by Currrupted** [comment]
Hello! I have recently started using floccus, it is very useful for me, I'm so glad something like this exists! However, I cannot really use it for all my bookmarks, as all bookmarks I have had locally use tags... It seems like there isn't any real development from Firefox' side... Would it be possible to use something like this? 
https://github.com/azappella/webextension-experiment-tags

**Comment by marcelklehr** [comment]
Hi @Currrupted 
Thank you for weighing in on this. The project you mentioned is only available for Nightly or unbranded Beta releases of Firefox, sadly. This will not help much, I think. Mozilla are working to improve things, as far as I can see, but one step at a time, as they have little resources.

**Comment by mortee** [comment]
I would like to see this implemented, too!

**Comment by dgudim** [comment]
Just set up syncing with floccus and saw that the tags are not syncchronised. And found this issue. I hope folks at Mozilla implement it sometime...

**Comment by redactedscribe** [comment]
The middle-ground I've found is to [append tags to bookmark titles](https://www.reddit.com/r/firefox/comments/qseq3u/is_there_any_way_to_export_bookmark_tags_to/) and then forget about Firefox tags altogether. The added benefit is that if you ever use / sync your bookmarks in / to Chrome, your pseudo-tags will come with them too (Chrome doesn't support tags). Plus, after adoption of this system, you can continue to append "tags" to bookmark titles ("My Bookmark [tag 1; tag 2]"). Cross-browser and most of the benefit.

**Comment by dgudim** [comment]
Yeah, I thought about that. One drawback with that is that if you are syncing with a system that supports tags, you will have 0 tags there with no autocomplete and other stuff.

Is it possible for floccus to maybe support this way (convert tags from titles into native tags for the sync platform and vice-versa)
